### ISSUE TITLE
Python bindings are not side-by-side installable?

### ISSUE THREAD

**Issue by traversaro** [gazebosim/gz-math#425]
## Environment
* OS Version: Ubuntu 22.04
* Source or binary build? Both binary and source, https://github.com/ignitionrobotics/ign-math/tree/ignition-math6_6.10.0 and https://github.com/ignitionrobotics/ign-math/commit/cd439fa1ac6b5b08bcb38e80453c84a579536112

## Description
* Expected behavior: One characterizing feature of Ignition libraries is that major version of it are side-by-syde installable. However, while working on https://github.com/conda-forge/libignition-math4-feedstock/issues/20 I noticed that the Python bindings install the file `$PREFIX/lib/python3.7/site-packages/ignition/math.cpython-37m-x86_64-linux-gnu.so`, that is not major version specific, that so will conflict with the same file installed by another major version. I am not sure if this change is intended (it may be documented somewhere, but I did not found anything) or not intentional. 
* Actual behavior:  Python bindings of major version side-by-side installable, or the fact that they are not side-by-side installable documented. 

## Steps to reproduce

Install  https://github.com/ignitionrobotics/ign-math/tree/ignition-math6_6.10.0 and https://github.com/ignitionrobotics/ign-math/commit/cd439fa1ac6b5b08bcb38e80453c84a579536112 in the same prefix. The latest one installed will overwrite the `$PREFIX/lib/python3.7/site-packages/ignition/math.cpython-37m-x86_64-linux-gnu.so` file.

**Comment by traversaro** [comment]
This can be shown also by installing the `python3-ignition-math6` and `python3-ignition-math7` nightly packages on a Ubuntu 22.04 system, it fails as the file `/usr/lib/python3/dist-packages/ignition/math.cpython-310-x86_64-linux-gnu.so` is in both packages:
~~~
straversaro@IITICUBLAP257:~$ sudo apt install python3-ignition-math6 python3-ignition-math7
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
python3-ignition-math6 is already the newest version (6.10.0-100~jammy).
The following NEW packages will be installed:
  libignition-math7 python3-ignition-math7
0 upgraded, 2 newly installed, 0 to remove and 1 not upgraded.
Need to get 0 B/743 kB of archives.
After this operation, 3,372 kB of additional disk space will be used.
Selecting previously unselected package libignition-math7:amd64.
(Reading database ... 110956 files and directories currently installed.)
Preparing to unpack .../libignition-math7_6.999.999+nightly+git20220505+1rbfcf56a60c42f5b5c96cf456a0d566958554219b-1~jammy_amd64.deb ...
Unpacking libignition-math7:amd64 (6.999.999+nightly+git20220505+1rbfcf56a60c42f5b5c96cf456a0d566958554219b-1~jammy) ...
Preparing to unpack .../python3-ignition-math7_6.999.999+nightly+git20220505+1rbfcf56a60c42f5b5c96cf456a0d566958554219b-1~jammy_amd64.deb ...
Unpacking python3-ignition-math7 (6.999.999+nightly+git20220505+1rbfcf56a60c42f5b5c96cf456a0d566958554219b-1~jammy) ...
dpkg: error processing archive /var/cache/apt/archives/python3-ignition-math7_6.999.999+nightly+git20220505+1rbfcf56a60c42f5b5c96cf456a0d566958554219b-1~jammy_amd64.deb (--unpack):
 trying to overwrite '/usr/lib/python3/dist-packages/ignition/math.cpython-310-x86_64-linux-gnu.so', which is also in package python3-ignition-math6 6.10.0-100~jammy
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 /var/cache/apt/archives/python3-ignition-math7_6.999.999+nightly+git20220505+1rbfcf56a60c42f5b5c96cf456a0d566958554219b-1~jammy_amd64.deb
needrestart is being skipped since dpkg has failed
E: Sub-process /usr/bin/dpkg returned an error code (1)
~~~

**Comment by traversaro** [comment]
This will be probably solved by https://github.com/gazebosim/gz-math/pull/503 .

**Comment by mjcarroll** [comment]
Closed by #503
